### PR TITLE
Add support for architecture

### PR DESCRIPTION
**Pull request by akhurana001**
SDK changes to support architecture for container/images : Ref: https://github.com/flyteorg/flyte/issues/1539

### CODE REVIEW

**Comment by pingsutw** [comment]
@akhurana001 Thank you for working on this.
Could you sign the commit? thanks!
https://github.com/flyteorg/flytekit/pull/752/checks?check_run_id=4231839088

**Comment by eapolinario** [comment]
The snyk error seems to be an issue on their side. According to https://support.snyk.io/hc/en-us/articles/360004358517-Unknown-PR-test-error it should be temporary and we can force a re-run by closing and opening the PR again.

**Comment by eapolinario** [comment]
Retrying snyk tests just once more.

**Comment by eapolinario** [comment]
Retrying snyk tests again.

File: flytekit/models/task.py [hunk 1]
```diff
@@ -854,10 +854,12 @@ def __init__(self, image, command, args, resources, env, config, data_loading_co
         :param Resources resources: A definition of requisite compute resources.
         :param dict[Text, Text] env: A definition of key-value pairs for environment variables.
         :param dict[Text, Text] config: A definition of configuration key-value pairs.
+        : param Architecture: Architecture supported by this container's image.
```
Comment (wild-endeavor) [inline flytekit/models/task.py:857]:
```suggestion
        :param architecture: Architecture supported by this container's image.
```

File: flytekit/models/task.py [hunk 2]
```diff
@@ -843,7 +843,7 @@ def from_flyte_idl(cls, pb2: _core_task.DataLoadingConfig) -> "DataLoadingConfig
 
 
 class Container(_common.FlyteIdlEntity):
-    def __init__(self, image, command, args, resources, env, config, data_loading_config=None):
+    def __init__(self, image, command, args, resources, env, config, architecture=None, data_loading_config=None):
```
Comment (wild-endeavor) [inline flytekit/models/task.py:846]: add type hint here? 
```suggestion
    def __init__(self, image, command, args, resources, env, config, architecture: typing.Optional[int] = None, data_loading_config=None):
```

File: flytekit/models/task.py [hunk 3]
```diff
@@ -872,6 +874,14 @@ def image(self):
         """
         return self._image
 
+    @property
+    def architecture(self):
```
Comment (wild-endeavor) [inline flytekit/models/task.py:878]:
```suggestion
    def architecture(self) -> int:
```
or `typing.Optional[int]`

File: flytekit/models/task.py [hunk 4]
```diff
@@ -872,6 +874,14 @@ def image(self):
...
+        """
+        Architecures supported by the container's image.
+        :rtype: Text
```
Comment (wild-endeavor) [inline flytekit/models/task.py:881]: shouldn't this be int?  Also we can just remove this line now that we have mypy style hints.  https://github.com/flyteorg/flyteidl/pull/226/files#diff-8bd2613bf781125f77247c29a2f5502658c63cccf2b597a9ec3695ef1e7b2ca9R188

File: flytekit/models/task.py (hunk 2 above)
Comment (wild-endeavor) [reply flytekit/models/task.py:846]: or maybe we should just make the default 0/unknown and always fill it in.  either way as long as it works with the backend.

File: flytekit/models/task.py (hunk 3 above)
Comment (reverson) [reply flytekit/models/task.py]: Done

File: flytekit/models/task.py (hunk 4 above)
Comment (reverson) [reply flytekit/models/task.py]: Updated with int

File: flytekit/models/task.py (hunk 2 above)
Comment (reverson) [reply flytekit/models/task.py]: Defaulted to 0, the backend changes check for 0/Unknown and don't do anything in that case, so this should work.

File: flytekit/models/task.py (hunk 1 above)
Comment (reverson) [reply flytekit/models/task.py]: Done

**Pull request merged**